pqarrow/builder/optbuilders.go

```diff
@@ -688,3 +705,9 @@ func (b *OptInt32Builder) ResetToLength(n int) {
 	b.data = b.data[:n]
 	b.validityBitmap = resizeBitmap(b.validityBitmap, n)
 }
+
+func (b *OptInt32Builder) ReserveToLength(n int) {
```

Comment:
can we just call this `Reserve` to be more in line with the arrow API?

Reply:
done c236674